packages/components/src/components/primitives/Link/Link.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { FC } from 'react';
+import {
+  Box,
+  Link as ChakraLink,
+  Text,
+  Flex,
+  FlexboxProps,
+} from '@chakra-ui/react';
+
+import { LinkProps } from './types';
+
+export const Link: FC<LinkProps> = ({
+  children,
+  icon,
+  iconPosition = 'trailing',
+  size = 'm',
+  ...props
+}) => {
+  const Icon: React.FC = () =>
+    React.cloneElement(icon, {
+      size,
+    });
+
+  const flexDirection = {
+    trailing: 'row-reverse',
+    leading: 'row',
+  }[iconPosition] as FlexboxProps['flexDirection'];
```

Comment:
I think we have to change `leading` to be `row-reverse` and trailing to be `row` given the order of the Text and Icon